### ISSUE TITLE
Review connection and result parsing in all connectors

### ISSUE THREAD

**Issue by geclos** [latitude-dev/latitude#138]
## TODO
- Review / implement SSL support
- QA with real data
- Review and fix the docs

## Progress

- [x] postgres
- [x] mysql
- [x] athena
- [x] bigquery 
- [x] clickhouse @samulatitude 
- [ ] databricks @samulatitude 
- [x] duckdb
- [x] redshift @samulatitude 
- [x] snowflake @geclos 
- [x] sqlite @geclos 
- [ ] trino

**Comment by PolGuixe** [comment]
I have tested a connection to a postgres DB from Heroku (hosted in AWS), and I am getting this error:

![Screenshot 2024-03-25 at 21 16 02](https://github.com/latitude-dev/latitude/assets/6626225/27c6375d-7a89-466d-9e59-822a87adb493)

source -not real db name, user, pwd -
![Screenshot 2024-03-25 at 21 16 27](https://github.com/latitude-dev/latitude/assets/6626225/76dc08a9-125f-4e4c-a5c4-cd542940cb13)

works fine with Postico and other PG clients from local

**Comment by geclos** [comment]
Hey @PolGuixe ,

## TL;DR
Do this:

``yaml
ssl:
   rejectUnauthorized: false
```

## Long explanation
This issue is really an ongoing initiative we have to QA all our db connectors. Your case should already be supported by our Postgres connector.

What is happening is that the default self-signed SSL certificates the postgres connector is using are not supported by your DB. A quick and easy way to fix this is to replace `ssl: true` with the following:

```yaml
ssl:
   rejectUnauthorized: false
```

This will maintain the SSL encryption but will tell your db to not check the certificate validity, which is fine in cases like yours where you control the db and is what most connectors do under the hood to simplify configuration for the user. If you want to be more strict and have the db confirm the certificate authority for your SSL connection you can include your OS's CA root store, like so:

```yaml
ssl:
  ca: /etc/ssl/cert.pem # this path will depend on your operating system
```

You can check a list of OSs and their CA paths in [our docs](https://docs.latitude.so/sources/postgresql#ca-root-configuration). 

A third solution involves using custom keys and certificates provided by your DB provider but I don't think it's worth getting into it since it's almost surely not your case.

**Comment by PolGuixe** [comment]
@geclos thanks! This has resolved the issue. 🎉
